src/states/node.rs

```diff
@@ -110,6 +110,8 @@ pub struct Node {
     routing_msg_backlog: Vec<RoutingMessage>,
     /// Cache of `OwnSectionMerge` messages we have received, by sender section prefix.
     merge_cache: LruCache<Prefix<XorName>, SectionMap>,
+    /// Cache of union of our merged section, deduced from multiple `OwnSectionMerge`.
+    our_merged_section_cache: LruCache<Prefix<XorName>, (bool, BTreeSet<XorName>)>,
```

Comment:
in case of merge after merge, the out-dated OwnSectoinMerge of first merge may overwrite the current on-going merge, if using Option only.
Is it possible?